clightning/config.go

```diff
@@ -95,6 +96,35 @@ func SetPeerswapPaths(plugin *glightning.Plugin) Processor {
 	}
 }
 
+// CheckForDeprecatedApiConfig tries to detect if allow-deprecated-apis is false
+// in the CLN config. If it is set false, we print a warning and exit because
+// deprecated CLN API fields might break PeerSwap.
+func CheckForDeprecatedApiConfig(client *ClightningClient) Processor {
+	return func(c *Config) (*Config, error) {
+		conf, err := client.glightning.ListConfigs()
+			if err != nil {
+				return nil, err
+			}
```

Comment:
Nit: I think one extra indent is added.
How about the following?

```suggestion
		if err != nil {
			return nil, err
		}
```